### PR TITLE
[16.0] [MIG] stock_lot_on_hand_first: Migration to 16.0

### PR DESCRIPTION
**Pull request by fvaqueriza**
Standard migration

### CODE REVIEW

File: stock_lot_on_hand_first/models/stock_move.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    display_lots_on_hand_first = fields.Boolean(
+        related="picking_type_id.display_lots_on_hand_first",
+        readonly=True,
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_move.py:13]: readonly is by default in relared fields

File: stock_lot_on_hand_first/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+
+    _inherit = "stock.picking"
+
+    display_lots_on_hand_first = fields.Boolean(
+        related="picking_type_id.display_lots_on_hand_first",
+        readonly=True,
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_picking.py:13]: readonly is by default in relared fields

File: stock_lot_on_hand_first/models/stock_lot.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+from odoo.osv.expression import AND
+
+
+class StockLot(models.Model):
+    _inherit = "stock.lot"
+    _rec_search_name = ["name", "code"]
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_lot.py:9]: WHY?

File: stock_lot_on_hand_first/models/stock_lot.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    @api.model
+    def _name_search(self, name="", args=None, operator="ilike", limit=80):
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_lot.py:12]: add name_get_uid=None parameter, is implement in base method

File: stock_lot_on_hand_first/models/stock_lot.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                # Since we cannot merge the Query objects to respect the intended order
+                # in which we want to display the results,  we return the list of ids
+                # that will be used by browse in the name_search implementation.
+                return (
+                    self.browse(with_quantity_ids).ids
+                    + self.browse(without_quantity_ids).ids
+                )
+
+        return super()._name_search(
+            name=name, args=args, operator=operator, limit=limit
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_lot.py:46]: add name_get_uid=name_get_uid parameter

File: stock_lot_on_hand_first/models/stock_lot.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+            with_quantity_domain = AND([args, [("product_qty", ">", 0)]])
+            with_quantity_count = self.env["stock.lot"].search_count(
+                with_quantity_domain
+            )
+
+            if with_quantity_count >= limit:
+                args = with_quantity_domain
+            else:
+                with_quantity_ids = super()._name_search(
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_lot.py:26]: add name_get_uid=name_get_uid parameter

File: stock_lot_on_hand_first/models/stock_lot.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                    name=name, args=with_quantity_domain, operator=operator, limit=limit
+                )
+                without_quantity_ids = super()._name_search(
```
Comment (RodrigoBM) [inline stock_lot_on_hand_first/models/stock_lot.py:29]: add name_get_uid=name_get_uid parameter

**Review by RodrigoBM — APPROVED** [review]
LGTM!

**Review by RodrigoBM — APPROVED** [review]

**Review by LuisNovalio — APPROVED** [review]
LGTM Functional review 

https://github.com/OCA/stock-logistics-workflow/assets/175136695/1d5deaba-f698-4bca-84c8-86763bf71b6a

**Comment by fvaqueriza** [comment]
Hi, could someone from @OCA/logistics-maintainers check this PR?

**Comment by AdrianaSaiz** [comment]
Good morning could someone from @OCA/logistics-maintainers check this PR?

**Comment by fvaqueriza** [comment]
Hi, good moring, could someone from @OCA/logistics-maintainers check this PR?

**Review by jbaudoux — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
LGTM

**Pull request merged**